### PR TITLE
Fix covs

### PR DESCRIPTION
**Pull request by alvarofe**

### CODE REVIEW

File: libr/parse/p/parse_6502_pseudo.c [hunk 1]
```diff
@@ -88,13 +84,14 @@ static int replace(int argc, const char *argv[], char *newstr, ADDR_TYPE type) {
...
+				} else {
 					newstr[k] = ops[i].str[j];
+				}
 			}
 			newstr[k] = '\0';
 			if (argc == 4 && argv[2][0] == '[') {
-				strcat(newstr + k, "+");
-				strcat(newstr + k + 3, argv[3]);
+				strcat (newstr + k, "+");
+				strcat (newstr + k + 3, argv[2]);
```
Comment (alvarofe) [inline libr/parse/p/parse_6502_pseudo.c]: check this @ricardoquesada please :). This function is called in `parse`. The maximum index for wa is 2 and in `replace` was using 3 causing a oob read. Check if this break something though running the regression test t.esil/6502 i get green
```
const char *wa[] = {w0, w1, w2};                                                                                                                                                      
int nw = 0;                                                                                                                                                                           
for (i = 0; i < 3; i++) {                                                                                                                                                             
             if (wa[i][0]) nw++;                                                                                                                                                           
}                                                                                                                                                                                     
replace (nw, wa, str, atype);
```

**Comment by radare** [comment]
It fails on the java (dsojson) stuff.

**Comment by alvarofe** [comment]
I'm looking at it now ;)

**Comment by alvarofe** [comment]
i don't know why it's failing because in my laptop i get green

https://asciinema.org/a/4avagm6lgwd4xdi3lq4dfo3qy

**Comment by ricardoquesada** [comment]
Apparently what broke `t.esil/6502` was this commit https://github.com/radare/radare2/commit/876a4052be544d2ca748b1f432ea6c3be0a0ce1c

I was using `aeim` and apparently I shouldn't do that. I'll use `aei` instead. I'll send a new PR with that.

@alvarofe I'm not the author of the `parse_6502_pseudo.c` code. In fact, I don't know how to use the parse functionality of r2, but if you tell me how to use it I can tell you if the output is correct.

**Comment by radare** [comment]
@ricardoquesada i have marked those tests as broken for the moment.

**Comment by radare** [comment]
Merged!

**Comment by ricardoquesada** [comment]
@radare ok. here is the PR that fixes those tests: https://github.com/radare/radare2-regressions/pull/233

**Comment by alvarofe** [comment]
@ricardoquesada sorry then I thought you were in charge of this.

**Pull request closed without merging**